### PR TITLE
Types (de)serialization tests

### PR DESCRIPTION
**Pull request by popzxc**
Some tests for RPC types (de)serialization wouldn't harm, would they?

### CODE REVIEW

**Review by dvdplm — APPROVED** [review]
This looks great, thank you very much. :)

File: types/src/v2/request.rs
Comment (niklasad1) [inline types/src/v2/request.rs:136]: not your fault but this shouldn't be parsed as a `JsonRpcRequest` instead should be parsed as `JsonRpcNotification`, #325 fixes this.

Comment (popzxc) [reply types/src/v2/request.rs:136]: Should I just remove it now, or wait until the linked PR is merged and implement a negative test?

Comment (niklasad1) [reply types/src/v2/request.rs:136]: I merged it #325, do mind merging/rebasing to master? Otherwise, I can help with it too.

Sorry for introducing this merge conflicts.

Comment (popzxc) [reply types/src/v2/request.rs]: No problem. I've rebased to master, removed tests that became duplicated (tried to leave ones that are more comprehensive), and aligned my tests with the surrounding code.

File: types/src/v2/request.rs
Comment (niklasad1) [inline types/src/v2/request.rs:94]: cool, we should implement `PartialEq` for the types manually that has a `serde_json::RawValue` field in another PR I guess.

**Review by niklasad1 — APPROVED** [review]

**Pull request merged**
